src/components/common/TokenAmountInput/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { Button, Divider, FormControl, InputLabel, MenuItem, TextField } from '@mui/material'
+import css from './styles.module.css'
+import NumberField from '@/components/common/NumberField'
+import { validateDecimalLength, validateLimitedAmount } from '@/utils/validation'
+import { AutocompleteItem } from '@/components/tx-flow/flows/TokenTransfer/CreateTokenTransfer'
+import { useFormContext } from 'react-hook-form'
+import { type BigNumber } from '@ethersproject/bignumber'
+import classNames from 'classnames'
+
+enum TokenAmountFields {
+  tokenAddress = 'tokenAddress',
+  amount = 'amount',
+}
+
+const TokenAmountInput = ({
+  balances,
+  selectedToken,
+  onMaxAmountClick,
+  maxAmount,
+}: {
+  balances: any[]
+  selectedToken: any
+  onMaxAmountClick: () => void
+  maxAmount: BigNumber
+}) => {
+  const {
+    formState: { errors },
+    register,
+    resetField,
+    watch,
+  } = useFormContext<{ [TokenAmountFields.tokenAddress]: string; [TokenAmountFields.amount]: string }>()
+
+  const tokenAddress = watch(TokenAmountFields.tokenAddress)
+  const isAmountError = !!errors[TokenAmountFields.tokenAddress] || !!errors[TokenAmountFields.amount]
+
+  return (
+    <FormControl className={classNames(css.outline, { [css.error]: isAmountError })} fullWidth>
+      <InputLabel shrink required className={css.label}>
+        {errors[TokenAmountFields.tokenAddress]?.message || errors[TokenAmountFields.amount]?.message || 'Amount'}
+      </InputLabel>
+      <div className={css.inputs}>
+        <NumberField
+          variant="standard"
+          InputProps={{
+            disableUnderline: true,
+            endAdornment: (
+              <Button className={css.max} onClick={onMaxAmountClick}>
+                MAX
+              </Button>
+            ),
+          }}
+          className={css.amount}
+          required
+          placeholder="0"
+          {...register(TokenAmountFields.amount, {
+            required: true,
+            validate: (val) => {
+              const decimals = selectedToken?.tokenInfo.decimals
+              return validateLimitedAmount(val, decimals, maxAmount.toString()) || validateDecimalLength(val, decimals)
+            },
+          })}
+        />
+        <Divider orientation="vertical" flexItem />
+        <TextField
+          select
+          variant="standard"
+          InputProps={{
+            disableUnderline: true,
+          }}
+          className={css.select}
+          {...register(TokenAmountFields.tokenAddress, {
+            required: true,
+            onChange: () => {
+              resetField(TokenAmountFields.amount, { defaultValue: '0' })
```

Comment:
This doesn't work for the case where the user selects a token, goes to the next step and back and changes the token address since the component is mounted again with the changed value instead of the default value.